wandb/__init__.pyi

```diff
@@ -416,48 +416,93 @@ def log(
+    to a previous step.
 
-    `wandb.log` is not intended to be called more than a few times per second.
-    If you want to log more frequently than that it's better to aggregate
-    the data on the client side or you may get degraded performance.
+    Note that you can use any metric as the X axis in charts.
+    In many cases, it is better to think of the W&B step as
+    a sort of timestamp rather than a training step.
+
+    ```
+    # Example: log an "epoch" metric for use as an X axis.
+    run.log({"epoch": 40, "train-loss": 0.5})
+    ```
+
+    See also `run.define_metric`.
+
+    It is possible to use multiple `log` invocations to log to
+    the same step with the `step` and `commit` parameters.
+    The following are all equivalent:
+
+    ```
+    # Normal usage:
+    run.log({"train-loss": 0.5, "accuracy": 0.8})
+    run.log({"train-loss": 0.4, "accuracy": 0.9})
+
+    # Implicit step without auto-incrementing:
+    run.log({"train-loss": 0.5}, commit=False)
+    run.log({"accuracy": 0.8})
+    run.log({"train-loss": 0.4}, commit=False)
+    run.log({"accuracy": 0.9})
+
+    # Explicit step:
+    run.log({"train-loss": 0.5}, step=current_step)
+    run.log({"accuracy": 0.8}, step=current_step)
+    current_step += 1
+    run.log({"train-loss": 0.4}, step=current_step)
+    run.log({"accuracy": 0.9}, step=current_step)
+    ```
 
     Arguments:
-        data: (dict, optional) A dict of serializable python objects i.e `str`,
-            `ints`, `floats`, `Tensors`, `dicts`, or any of the `wandb.data_types`.
-        commit: (boolean, optional) Save the metrics dict to the wandb server
-            and increment the step.  If false `wandb.log` just updates the current
-            metrics dict with the data argument and metrics won't be saved until
-            `wandb.log` is called with `commit=True`.
-        step: (integer, optional) The global step in processing. This persists
-            any non-committed earlier steps but defaults to not committing the
-            specified step.
-        sync: (boolean, True) This argument is deprecated and currently doesn't
-            change the behaviour of `wandb.log`.
+        data: A `dict` with `str` keys and values that are serializable
+            Python objects including: `int`, `float` and `string`;
+            any of the `wandb.data_types`; lists, tuples and NumPy arrays
+            of serializable Python objects; other `dict`s of this
+            structure.
+        step: The step number to log. If `None`, then an implicit
+            auto-incrementing step is used. See the notes in
+            the description.
+        commit: If true, finalize and upload the step. If false, then
+            accumulate data for the step. See the notes in the description.
+            If `step` is `None`, then the default is `commit=True`;
+            otherwise, the default is `commit=False`.
+        sync: This argument is deprecated and does nothing.
```

Comment:
do you think we can just remove it silently? given that it’s never used?

Reply:
We have deprecation telemetry for it, could we check that first? Just in case there are unmaintained libraries out there that are passing `sync` but get used with new `wandb` versions.

Reply:
Perfect! Love this approach